pkg/server/server.go

```diff
@@ -363,11 +414,26 @@ func (s *Server) Check(ctx context.Context, req *openfgav1.CheckRequest) (*openf
 
 	ctx = typesystem.ContextWithTypesystem(ctx, typesys)
 
-	checkResolver := graph.NewLocalChecker(
+	var checkResolver graph.CheckResolver
+	localCheckResolver := graph.NewLocalChecker(
 		storagewrappers.NewCombinedTupleReader(s.datastore, req.ContextualTuples.GetTupleKeys()),
 		graph.WithResolveNodeBreadthLimit(s.resolveNodeBreadthLimit),
 		graph.WithMaxConcurrentReads(s.maxConcurrentReadsForCheck),
 	)
+	checkResolver = localCheckResolver
+
+	if s.checkCache != nil {
+		cachedCheckResolver := graph.NewCachedCheckResolver(
+			localCheckResolver,
+			graph.WithExistingCache(s.checkCache),
+			graph.WithCacheTTL(s.checkQueryCacheTTL),
+		)
+
+		localCheckResolver.SetDelegate(cachedCheckResolver)
```

Comment:
Echoing https://auth0.slack.com/archives/C2ZCK92Q6/p1692227188656589

This is like a brain teaser and we shouldn't have those. Is there any way we can make this simpler please?

Reply:
@jpadilla what do you think? since you are a newcomer to the code. Is this easy to follow?

Reply:
Hopefully after this clean up is easier to read and understand